### DISCUSSION TITLE
SD XL

### DISCUSSION THREAD

**Discussion by aziRna** [AIrjen/OneButtonPrompt#120] [Ideas]
Hey i have been using your extention for too long and i started to do somethings on SD XL
i thought it can be really good for you to make a extention for SD XL?
Because im generating prompt and pasting it to into SD XL 😆

**Comment by AIrjen** [comment]
I am not sure what you mean by supporting SD XL. SD XL is a model. One button prompt currently supports tools, such as Automatic1111 and ComfyUI. Those tools can support the model, such as SD XL.

**Reply by Pokis** [reply]
and yes and no. Even though you can use it, you cannot use it to the fullest, as SDXL models can take 4 different prompts,
subject positive, style positive, subject negative and style negative. Whileas onebuttonprompt only provides single positive prompt. As a minimum it would require to provide a positive subject and positive style prompts for good effect, negatives are not as important.  

Currently it has them both as one output, and there is no real split symbol or anything that could be used for splitting between the two. If we could make as output:
* Concatenated (currently existing)
* Positive subject (same but subject oriented data points)
* Positive style (same but style oriented data points)

It would cover most of the flows imho.

edit: TBH we can concatenate ourselves, rather positive subject and positive style should be sufficient

**Reply by AIrjen** [reply]
Hi Pokis,

That is an interesting idea. This should theoretically be possible to sort out the subject from the stylings. There is already a subject only mode build in. I just have to figure out how to split it correctly.

I will have a think about this, I think the only place it makes sense to add is ComfyUI (since most other tools only have 1 prompt field) and don't use the double SDXL prompt fields.

**Reply by AIrjen** [reply]
This is available on comfyUI now:

![image](https://github.com/AIrjen/OneButtonPrompt/assets/130234949/f7219ec8-9d6b-436d-8028-c098204e4f2a)

I don't really like it, but it works :) You can do some things with it, such as just taking prompt_l (stylings) do to other things. So, go on, and have fun :)
